CMakeLists.txt

```diff
@@ -314,6 +318,25 @@ option(NO_DX "Disable DirectX support")
 option(NO_TESTS "Disable all tests")
 option(NO_GLTESTS "Disable GL tests")
 
+option(MSVC_STATIC_CRT_RELEASE "Statically link MSVC CRT in Release" OFF)
+option(MSVC_STATIC_CRT_DEBUG "Statically link MSVC CRT in Debug" OFF)
+
+if(MSVC AND MSVC_STATIC_CRT_RELEASE)
+    message(STATUS "Using static MSVC CRT Release")
+    # http://stackoverflow.com/a/32128977/486990
+    add_compile_options(
+        "$<$<CONFIG:RelWithDebInfo>:/MT>"
+        "$<$<CONFIG:Release>:/MT>"
+        "$<$<CONFIG:MinSizeRel>:/MT>"
+    )
+endif()
+if(MSVC AND MSVC_STATIC_CRT_DEBUG)
```

Comment:
Gotcha. Makes sense. All though, I'm not entirely sure how to set up CMake to do that.
That means adding additional configs so you'd have `Release`, `Release Static`, `Debug`, `Debug Static` etc?

There's another aspect of this as well, 32bit vs 64bit. At the moment I am working on an application plugin which has both 64bit and 32bit builds. I've yet to find a way to make CMake generate a single solution that include both versions - like you'd normally have when manually crafting.

My current way around this is to create two build directories `build32` and build64` and then trigger CMake and msbuild. If there is a way to generate all in one solution that'd be great.

Btw, having all these CRT configs in one file - can this be done in a separate issue? I agree that it's be really convenient, but at the moment I'm not familiar with how to do that - and not having static CRT is somewhat of a blocker for my project.